mm2src/coins/utxo/utxo_common.rs

```diff
@@ -3733,6 +3851,25 @@ where
     }
 }
 
+/// Sorts and deduplicates the given `unspents` in ascending order.
+fn sort_dedup_unspents<I>(unspents: I) -> Vec<UnspentInfo>
+where
+    I: IntoIterator<Item = UnspentInfo>,
+{
+    unspents
+        .into_iter()
+        // dedup just in case we add duplicates of same unspent out
+        .unique_by(|unspent| unspent.outpoint.clone())
```

Comment:
As per https://doc.rust-lang.org/stable/std/vec/struct.Vec.html#method.dedup_by, “ If the vector is sorted, this removes all duplicates”. You can sort this first and then safely dedup without additional cloning.

Reply:
> If the vector is sorted, this removes all duplicates.

It's true if the `Vec::dedup_by` method is used for the same key as the vector was sorted. But we first need to sort by `value` and then dedup by `outpoint`, so there can be a bug:
What if
```rust
let unspents = vec![
  UnspentOutput { value: 10000, outpoint: "OutPoint1" },
  UnspentOutput { value: 5000, outpoint: "OutPoint2" },
  UnspentOutput { value: 10000, outpoint: "OutPoint3" },
  UnspentOutput { value: 10000, outpoint: "OutPoint1" },
];
```
When we sort the items by `value`, we can get an iterator in the following order:
```rust
let unspents = vec![
  UnspentOutput { value: 5000, outpoint: "OutPoint2" },
  UnspentOutput { value: 10000, outpoint: "OutPoint1" },
  UnspentOutput { value: 10000, outpoint: "OutPoint3" },
  UnspentOutput { value: 10000, outpoint: "OutPoint1" },
];
```
And now `Vec::dedup_by` will not remove duplicates.

Reply:
Excuse me, I didn’t pay attention that keys are different. But I’m still interested if it’s possible to avoid cloning here 🙂

Reply:
>  But I’m still interested if it’s possible to avoid cloning here

Unfortunately, it's not possible.
```
error: lifetime may not live long enough
    --> mm2src/coins/utxo/utxo_common.rs:3860:30
     |
3860 |         .unique_by(|unspent| &unspent.outpoint)
     |                     -------- ^^^^^^^^^^^^^^^^^ returning this value requires that `'1` must outlive `'2`
     |                     |      |
     |                     |      return type of closure is &'2 chain::OutPoint
     |                     has type `&'1 utxo::rpc_clients::UnspentInfo`

```
Also `OutPoint` doesn't implement `Copy` trait.

Reply:
`OutPoint` contains H256 (which implements copy) and u32, we can derive Copy for Outpoint too.

Reply:
Done